agent/client/client.go

```diff
@@ -144,3 +154,32 @@ func (c *Client) getName() (string, error) {
 func isCancelledError(err error) bool {
 	return err != nil && strings.Contains(err.Error(), "context canceled")
 }
+
+func (c *Client) reconnect() error {
+	// connection is not working. We need to reconnect
+	err := retry.Do(func() error {
+		return c.connect(context.Background())
+	}, retry.Attempts(3), retry.Delay(1*time.Second))
```

Comment:
this will do 3 attempts with 1sec between them, right? Is that correct? Maybe we want to make it longer? like at least 1 min total?

Reply:
Thats delay doubles every attempt. So that would be 7 seconds total. If we set attempts as 6 we would get a bit longe than 1 minute